x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/rules/install_all.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IKibanaResponse, Logger } from '@kbn/core/server';
+import { buildRouteValidationWithZod } from '@kbn/zod-helpers';
+import type { RuleMigrationToInstall } from '../../../../../../common/siem_migrations/model/rule_migration.gen';
+import { canInstallMigrationRule } from '../../../../../../common/siem_migrations/utils';
+import { SIEM_RULE_MIGRATION_INSTALL_ALL_PATH } from '../../../../../../common/siem_migrations/constants';
+import type { InstallAllMigrationRulesResponse } from '../../../../../../common/siem_migrations/model/api/rules/rule_migration.gen';
+import { InstallAllMigrationRulesRequestBody } from '../../../../../../common/siem_migrations/model/api/rules/rule_migration.gen';
+import type { SecuritySolutionPluginRouter } from '../../../../../types';
+import { withLicense } from '../util/with_license';
+import { installRules } from './helpers';
+
+export const registerInstallAllMigrationRulesRoute = (
+  router: SecuritySolutionPluginRouter,
+  logger: Logger
+) => {
+  router.versioned
+    .post({
+      path: SIEM_RULE_MIGRATION_INSTALL_ALL_PATH,
+      access: 'internal',
+      security: { authz: { requiredPrivileges: ['securitySolution'] } },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: buildRouteValidationWithZod(InstallAllMigrationRulesRequestBody),
+          },
+        },
+      },
+      withLicense(
+        async (context, req, res): Promise<IKibanaResponse<InstallAllMigrationRulesResponse>> => {
+          const { migrationId } = req.body;
+
+          try {
+            const ctx = await context.resolve(['core', 'alerting', 'securitySolution']);
+
+            const ruleMigrationsClient = ctx.securitySolution.getSiemRuleMigrationsClient();
+            const soClient = ctx.core.savedObjects.client;
+            const rulesClient = ctx.alerting.getRulesClient();
+
+            const migrationRules = await ruleMigrationsClient.data.rules.get(migrationId);
```

Comment:
If we add a `filters?: QueryDslQueryContainer[]` option parameter to the `get` function. We'll to retrieve the _installable_ rules only from ES:
```
const filters: QueryDslQueryContainer[] = [
  { term: { translation_result: 'full' } },
  {
    nested: {
      path: 'elastic_rule',
      query: { bool: { must_not: { exists: { field: 'elastic_rule.id' } } } },
    },
  },
];

ruleMigrationsClient.data.rules.get(migrationId, { filters });
```

This way we won't have to iterate all migrations in Kibana.

Reply:
yup, that's exactly what I have on my pagination branch which I will add after we finish with this one :-)

Reply:
done